src/sqlparser/src/parser.rs

```diff
@@ -5574,6 +5582,105 @@ impl Parser<'_> {
         })
     }
 
+    fn parse_privilege_object_types(&mut self) -> ModalResult<PrivilegeObjectType> {
+        let object_type = if self.parse_keyword(Keyword::TABLES) {
+            PrivilegeObjectType::Tables
+        } else if self.parse_keyword(Keyword::SOURCES) {
+            PrivilegeObjectType::Sources
+        } else if self.parse_keyword(Keyword::SINKS) {
+            PrivilegeObjectType::Sinks
+        } else if self.parse_keywords(&[Keyword::MATERIALIZED, Keyword::VIEWS]) {
+            PrivilegeObjectType::Mviews
+        } else if self.parse_keyword(Keyword::VIEWS) {
+            PrivilegeObjectType::Views
+        } else if self.parse_keyword(Keyword::FUNCTIONS) {
+            PrivilegeObjectType::Functions
+        } else if self.parse_keyword(Keyword::SECRETS) {
+            PrivilegeObjectType::Secrets
+        } else if self.parse_keyword(Keyword::CONNECTIONS) {
+            PrivilegeObjectType::Connections
+        } else if self.parse_keyword(Keyword::SUBSCRIPTIONS) {
+            PrivilegeObjectType::Subscriptions
+        } else if self.parse_keyword(Keyword::SCHEMAS) {
+            PrivilegeObjectType::Schemas
+        } else {
+            return self.expected("TABLES, SOURCES, SINKS, MATERIALIZED VIEWS, VIEWS, FUNCTIONS, SECRETS, CONNECTIONS, SUBSCRIPTIONS or SCHEMAS");
```

Comment:
I know there is no privilege of index in the grant syntax, but what's the default privilege we should grant an index to a user? The same table of it should be considered as a materialzied view. Current implementation seems we follow the latter one.

Reply:
Nice catch! Yes, The same table of it will be considered as a materialized view. For standard `grant` and `revoke` privileges, when a table is granted or revoked `select` privilege, the index tables (including internal tables) will be applied with the corresponding operation. For default privilege of index, we should also follow this behavior. I will fix it.